### PR TITLE
Added test case for report polling

### PR DESCRIPTION
**Pull request by shreyanshss7**

### CODE REVIEW

File: tests/test_report_approval.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+    mock_report_details = {
+        'id': 'mock-user-id',
+        'org_id': 'orwruogwnngg',
+        'created_at': '2020-06-01T13:14:54.804+00:00',
+        'updated_at': '2020-06-11T13:14:55.201598+00:00',
+        'user_id': 'mock-user-id',
+        'user': mock_user_details,
+        'purpose': 'Business trip to London',
+        'currency': 'INR',
```
Comment (abhishek1234321) [inline tests/test_report_approval.py:52]: test with USD instead?

File: tests/test_report_approval.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        'amount': 47.99,
+        'tax': 18.23,
+        'state': 'APPROVER_PENDING',
+        'num_expenses': 3,
+        'is_manually_flagged': True,
```
Comment (abhishek1234321) [inline tests/test_report_approval.py:57]: manual flag happens only when an approver or admin flags it during report approval, so why test for this?

File: tests/test_report_approval.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        'is_policy_flagged': True,
+        'reimbursed_at': '2020-06-11T13:14:55.201598+00:00',
```
Comment (abhishek1234321) [inline tests/test_report_approval.py:59]: this will not be set at approval stage no?

File: tests/test_report_approval.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+        'approved_at': '2020-06-11T13:14:55.201598+00:00',
```
Comment (abhishek1234321) [inline tests/test_report_approval.py:60]: are you testing report approval?

File: tests/test_report_approval.py [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        'submitted_at': '2020-06-11T13:14:55.201598+00:00',
+        'claim_number': 'C/2021/02/R/907',
+        'source': 'string',
```
Comment (abhishek1234321) [inline tests/test_report_approval.py:63]: shouldn't mock data have some resemblance to actual stuff?

File: tests/test_report_approval.py [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+        'approvals': 'cs.[{{ "approver_user_id": {}, "state": "APPROVAL_PENDING" }}]'.format(mock_report_polling_detail.slack_user.fyle_user_id),
+        'last_submitted_at': 'gte.{}'.format(last_submitted_at),
+
+        # Mandatory query params required by sdk
+        'limit': 50, # Assuming no more than 50 reports will be there in 10 min poll
+        'offset': 0,
+        'order': 'last_submitted_at.desc'
+    }
+
+    mock_report_details = {
```
Comment (abhishek1234321) [inline tests/test_report_approval.py:44]: shouldn't we have a certain folder only for writing these mock objects? what does django or the community recommend?

**Review by abhishek1234321 — CHANGES_REQUESTED** [review]

File: tests/test_report_approval.py [hunk 7]
```diff
@@ -0,0 +1,115 @@
+import mock
```
[the remark this answers is not in the document]
Comment (shreyanshss7) [reply tests/test_report_approval.py:1]: It a convention of pytest, it recognises test files and test functions to run only if there is a prefix `test_<name>` in files and test functions

File: tests/test_report_approval.py (hunk 6 above)
Comment (shreyanshss7) [reply tests/test_report_approval.py:44]: Yes we should, there aren't much use case of reports data so kept it here only.

File: tests/test_report_approval.py (hunk 1 above)
Comment (shreyanshss7) [reply tests/test_report_approval.py:52]: Currency doesn't matter no sir this is just mock test to test the polling function.

File: tests/test_report_approval.py (hunk 2 above)
Comment (shreyanshss7) [reply tests/test_report_approval.py:57]: Test is not for this sir this is just a mock api response, it doesn't matter if it is false or true.

File: tests/test_report_approval.py (hunk 4 above)
Comment (shreyanshss7) [reply tests/test_report_approval.py:60]: This test is for report polling sir, approval test case PR will be separate

File: tests/test_report_approval.py (hunk 3 above)
Comment (shreyanshss7) [reply tests/test_report_approval.py:59]: mock data sir, doesn't matter in polling test case.

File: tests/test_report_approval.py [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+    fyle_report_approval.get_approver_reports.assert_called_with(mock_user, mock_query_params)
+
+    fyle_utils.get_fyle_report_url.assert_called()
+    fyle_utils.get_fyle_report_url.assert_called_with(mock_user.fyle_refresh_token)
+
+    slack_utils.get_user_display_name.assert_called()
+    slack_utils.get_user_display_name.assert_called_with(slack_client(), mock_user_details)
+
+    report_approval_messages.get_report_approval_notification.assert_called()
+    report_approval_messages.get_report_approval_notification.assert_called_with(mock_report_details, mock_user_display_name, mock_report_url)
```
Comment (shreyanshss7) [inline tests/test_report_approval.py:115]: Apologies forgot about this PR and test cases 🙁

File: tests/test_report_approval.py [hunk 9]
```diff
@@ -0,0 +1,115 @@
+import mock
+
```
Comment (shreyanshss7) [inline tests/test_report_approval.py:2]: Apologies forgot about this PR and test cases 🙁

File: tests/conftest.py [hunk 10]
```diff
@@ -0,0 +1,14 @@
+import json
```
Comment (shreyanshss7) [inline tests/conftest.py:1]: Apologies forgot about this PR and test cases 🙁

File: tests/conftest.py [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+import pytest
+
+
+@pytest.fixture
+def mock_fyle():
+    mock_data = open('tests/mock_data.json')
+    mock_data = json.load(mock_data)
+
+    fyle = mock.Mock()
+    fyle.approver.reports.list.return_value = mock_data['approver']['reports']['list']
```
Comment (shreyanshss7) [inline tests/conftest.py:12]: This convention is inline with our platform sdk `fyle.approver.reports.list.return_value`, meaning this is the methods and attributes we call to fetch data

File: tests/conftest.py (hunk 10 above)
Comment (shreyanshss7) [reply tests/conftest.py:1]: Update:- Added test cases for both report polling and report approve.

**Pull request closed without merging**